### PR TITLE
DevHub Xamarin.Forms example

### LINKED ISSUE #5: Xamarin Mobile Example Application
Build an example application to help developers understand how to build Mobile applications using Xamarin on the SAFE Network.

**The application must demonstrate the following features.**
- Installing latest [MaidSafe.SafeApp NuGet package](https://www.nuget.org/packages/MaidSafe.SafeApp/).
- Using Mock and Non-Mock features from the package
- Authentication
   - Mock authentication using MockAuth feature shipped in the package
   - Test net authentication using Safe Authenticator App
- Mutable Data Operation
   - CRUD Operations on Mutable Data
- App Custom URL Register (to receive the response from Safe Authenticator App)
   - Intent-Filter on Android
   - Custom URL on iOS

**Example App Must Include.**
- CI Setup
- Unit Tests
- Code analyzer (linter)

### LINKED ISSUE #11: Authentication Page UI/UX improvements
**Update Authentication Page to:**
- Show only one button for Authentication (mock/non-mock auth button)
- Add app build information showing whether the app is using mock/non-mock libs.

### LINKED ISSUE #12: UI/UX improvements in Safetodo example app
**Improve Main Page to**
    * Add option to mark a todo item as complete. 
    * Remove long item context menu and add edit, delete, complete option in card view.
    * Remove item description text from listview and show todo item description on the item details page.

**Add/Update Item Page Improve ments**
    * Replace add/edit popup dialog with a page.

### PR DESCRIPTION
**Pull request by ravinderjangra**
closes #5, closes #11, closes #12 

**SafeToDo Xamarin.Forms example App** 

**Purpose**:  Developers can use this app as a reference while using DevHub Xamarin guide.

**Features**:
- Mock authentication process (Using random account credentials).
- Test network authentication flow with SafeAuthenticator mobile app (using SAFE account).
- Using IntentFilter (Android), Custom URL (iOS) to send the AuthReq and get AuthResponse from mobile authenticator and initialize Session.
- Create and retrieve a RandomPrivateMutableData on the network (Created MData has used a todo list).
- Add/Retrieve entries to created mutable data (Add/Retrieve todo items from list).
- Update/Delete entries from existing mutable (Update/Delete todo items from the list).

### CODE REVIEW

File: SafeTodoExample/SafeTodoExample/SafeTodoExample.Android/Assets/AboutAssets.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Any raw assets you want to be deployed with your application can be placed in
+this directory (and child directories) and given a Build Action of "AndroidAsset".
+
+These files will be deployed with you package and will be accessible using Android's
```
Comment (krishnaIndia) [inline SafeTodoExample/SafeTodoExample/SafeTodoExample.Android/Assets/AboutAssets.txt]: Is this class needed? Are we using it anywhere?

**Comment by krishnaIndia** [comment]
Can we add CI support?

File: SafeTodoExample/SafeTodoExample/SafeTodoExample.Android/Assets/AboutAssets.txt (hunk 1 above)
Comment (ravinderjangra) [reply SafeTodoExample/SafeTodoExample/SafeTodoExample.Android/Assets/AboutAssets.txt]: It can be removed.

**Comment by ravinderjangra** [comment]
Yes, we can add CI but that will only cover build part, no tests

**Comment by ravinderjangra** [comment]
This repo has only 2 basic example apps. One desktop app (console) and another for Xamarin.Forms mobile app (android, iOS). These example apps are used as reference in DevHub guide to help developer's to understand the very basic data operations they can perform on SAFE network. Also these example apps will help them to understand the authentication process. 

Adding a CI will only have the build part with no tests. So, I don't think CI is required for this repo. And in case of any package update we can push the changes after testing on our machine.

**Comment by ravinderjangra** [comment]
@ashwinkumar6 will add that script in another pr.

**Pull request merged**